### PR TITLE
PyramidBox for face detection.

### LINKED ISSUE #920: Develop PyramidBox model for face detection.
- Add PyramidBox model for face detection:  https://arxiv.org/abs/1803.07737

### PR DESCRIPTION
**Pull request by qingqing01**
Fix #920

### CODE REVIEW

**Review by baiyfbupt — APPROVED** [review]
LGTM

File: fluid/face_detction/utility.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+"""Contains common utility functions."""
```
Comment (kuke) [inline fluid/face_detction/utility.py]: Please move this line to the bottom of the license. Or should we remove the license?

File: fluid/face_detction/pyramidbox.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+        self.ssh_conv5_norm = self._l2_norm_scale(self.ssh_conv5)
+
+        def permute_and_reshape(input, last_dim):
+            trans = fluid.layers.transpose(input, perm=[0, 2, 3, 1])
+            new_shape = [
+                trans.shape[0], np.prod(trans.shape[1:]) / last_dim, last_dim
+            ]
+            return fluid.layers.reshape(trans, shape=new_shape)
+
+        face_locs = []
```
Comment (kuke) [inline fluid/face_detction/pyramidbox.py]: Can merge every two lines

File: fluid/face_detction/train.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    values = [lr, lr * 0.5, lr * 0.25, lr * 0.1, lr * 0.01]
+    optimizer = fluid.optimizer.RMSProp(
+        learning_rate=fluid.layers.piecewise_decay(boundaries, values),
+        regularization=fluid.regularizer.L2Decay(0.00005), )
+
+    optimizer.minimize(loss)
+    place = fluid.CUDAPlace(0) if args.use_gpu else fluid.CPUPlace()
+    exe = fluid.Executor(place)
+    exe.run(fluid.default_startup_program())
+    #print(fluid.default_main_program())
```
Comment (kuke) [inline fluid/face_detction/train.py:40]: Can remove these lines

File: fluid/face_detction/pyramidbox.py [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+        self.ssh_conv3_norm = self._l2_norm_scale(self.ssh_conv3)
+        self.ssh_conv4_norm = self._l2_norm_scale(self.ssh_conv4)
+        self.ssh_conv5_norm = self._l2_norm_scale(self.ssh_conv5)
+
+        def permute_and_reshape(input, last_dim):
+            trans = fluid.layers.transpose(input, perm=[0, 2, 3, 1])
+            new_shape = [
+                trans.shape[0], np.prod(trans.shape[1:]) / last_dim, last_dim
+            ]
+            return fluid.layers.reshape(trans, shape=new_shape)
```
Comment (kuke) [inline fluid/face_detction/pyramidbox.py:179]: Is the shape of `trans` constant in runtime? otherwise we need to set shape via input.

**Review by kuke — COMMENTED** [review]
Almost LGTM

File: fluid/face_detction/pyramidbox.py (hunk 4 above)
Comment (qingqing01) [reply fluid/face_detction/pyramidbox.py:179]: Yeah, the shape of `trans` is constant.

File: fluid/face_detction/pyramidbox.py (hunk 2 above)
Comment (qingqing01) [reply fluid/face_detction/pyramidbox.py]: Done. Thanks!

File: fluid/face_detction/train.py (hunk 3 above)
Comment (qingqing01) [reply fluid/face_detction/train.py:40]: These code will be rewrite by tangxu to add the reader and train logic.

File: fluid/face_detction/utility.py (hunk 1 above)
Comment (qingqing01) [reply fluid/face_detction/utility.py]: Done. remove these lines.

**Pull request merged**